
Hudson–Kreitman–Aguade(HKA) test, which compares the relative levels of nucleotide diversity among loci, calibrated against divergence.
I will list several papers that is really useful for understanding the basic theory of hka test. 

1.Y. Ma, J. Wang, Q. Hu, J. Li, Y. Sun, L. Zhang, et al.Ancient introgression drives adaptation to cooler and drier mountain habitats in a cypress species complex. Communications Biology 2019 Vol. 2 Issue 1 Pages 213. DOI: 10.1038/s42003-019-0445-z.

2.Nourmohammad A, Rambeau J, Held T, Kovacova V, Berg J, Lässig M. Adaptive Evolution of Gene Expression in Drosophila. Cell Rep. 2017 Aug 8;20(6):1385-1395. doi: 10.1016/j.celrep.2017.07.033. PMID: 28793262.

3.Tetteh K K A, Stewart L B, Ochola L I, et al. Prospective identification of malaria parasite genes under balancing selection[J]. PloS one, 2009, 4(5): e5568.

at the beginning, I am so confused by the equation (if Polymophism/Divergence for target loci = Polymophism/Divergence for neutral loci) for detecting positive selection, because there is a significant skeww for this equation when balancing selection occurred due to increasing P and dereasing D. I think the right way is to find loci with (Polymophism/Divergence for target loci)/Polymophism/Divergence for neutral loci) > 1. However, this lab already has published numerious papers related to hka test for positive selection. I am afraid of that I will make a stupid decision. Subsequently, I read some research papers that use hka to detect balancing selection to answer my confusion. Finally,
Of course, you need read more if you want to avoid analysis mistakes
